modules/apps/object/object-web/src/main/resources/META-INF/resources/object_definitions/object_definition/details.jsp

```diff
@@ -41,6 +41,10 @@ renderResponse.setTitle(LanguageUtil.format(request, "edit-x", objectDefinition.
 	<aui:input name="<%= Constants.CMD %>" type="hidden" value="<%= Constants.UPDATE %>" />
 	<aui:input name="redirect" type="hidden" value="<%= currentURL %>" />
 
+	<c:if test="<%= objectDefinition.isApproved() || objectDefinition.isSystem() %>">
+		<aui:input name="enableObjectEntryHistory" type="hidden" value="<%= objectDefinition.isEnableObjectEntryHistory() %>" />
+	</c:if>
+
```

Comment:
@rodrigopaulino so this means that if the other input is disabled and the current value is true, when I try to save the object definition again the value to be saved will be false just because it's disabled? So we'll have issues with some other properties that have similar behavior, like enableComments, right?

Wouldn't it be better to solve this in the `ObjectDefinitionLocalServiceImpl` by moving the `objectDefinition.setEnableObjectEntryHistory(enableObjectEntryHistory);` 
to be after the 
`if (objectDefinition.isApproved()) {` 
in [here](https://github.com/liferay/liferay-portal/blob/4cd7d67353db4934b1cd131b64931d25684597e2/modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java#L1186-L1193) ? 

Same behavior as `objectDefinition.setName(name);`

Reply:
Hi, @carolmariaabb. Thanks for your inputs and perfect analysis of the consequences. As I understand you brought three different questions: two concerns and one alternative solution. I'll start with the concerns:
> @rodrigopaulino so this means that if the other input is disabled and the current value is true, when I try to save the object definition again the value to be saved will be false just because it's disabled?

That's correct. Liferay forces a checkbox parameter to have value [here](https://github.com/liferay/liferay-portal/blob/c197f39756a2fe2b946f2425f55e62b83a194985/portal-impl/src/com/liferay/portlet/internal/PortletRequestImpl.java#L1253-L1256).

> So we'll have issues with some other properties that have similar behavior, like enableComments, right?

It's possible, yes. Although I don't see this being a problem for any other toggle switch in this page, because they can't be turned on then become disabled. Let's take the "active/inactive" switch, for example: it's only disabled when the object definition is in draft or when it's a system object definition. When you create the object definition, the value will always be false, so when this switch becomes enabled, it can be turned on or off whenever the admin wants. Just an addendum, all these other switches check the permission for updating the object definition, but I don't see this being relevant since, even if the user could enter this page without permission, there's no way they could submit something. But I digress... moving on!

> Wouldn't it be better to solve this in the ObjectDefinitionLocalServiceImpl by moving the objectDefinition.setEnableObjectEntryHistory(enableObjectEntryHistory);
to be after the
if (objectDefinition.isApproved()) {
in [here](https://github.com/liferay/liferay-portal/blob/4cd7d67353db4934b1cd131b64931d25684597e2/modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java#L1186-L1193) ?

We could, but we wouldn't give the API users feedback on their bad attempt. Should we just ignore all invalid submissions and let the developer think they successfully updated their definition? Btw, this is not rhetorical. I'm really asking the protocol here 😆 

Hope these answer your questions.
Let me know anything!

Reply:
> That's correct. Liferay forces a checkbox parameter to have value [here](https://github.com/liferay/liferay-portal/blob/c197f39756a2fe2b946f2425f55e62b83a194985/portal-impl/src/com/liferay/portlet/internal/PortletRequestImpl.java#L1253-L1256).

Interesting.. Thanks for letting me know.

> It's possible, yes. Although I don't see this being a problem for any other toggle switch in this page,

:pray: 

> We could, but we wouldn't give the API users feedback on their bad attempt. Should we just ignore all invalid submissions and let the developer think they successfully updated their definition? 

Aren't we already giving this feedback via the `_validateEnableObjectEntryHistory` method in the `ObjectDefinitionLocalServiceImpl`?  If so, I don't think we need to try to change the value before the `if (objectDefinition.isApproved()) ` since the user will not be able to change anyway when the object is approved. It makes sense?

Reply:
> Aren't we already giving this feedback via the _validateEnableObjectEntryHistory method in the ObjectDefinitionLocalServiceImpl?

Ok. I get what you were saying now. I assumed you were asking to not validate the `enableObjectEntryHistory` if the definition was already approved. So, if we are validating then, wouldn't this service receive the request's forced false by the `PortletRequestImpl`? 
Imagine that the switch is turned on when the definition is in draft then, after publishing, we submit another change to the definition. The service would receive a `false` value for the argument `enableObjectEntryHistory` then, in comparison with the current value that is `true`, the validation would consider that we're trying to change this attribute after the definition was already published. Do you agree?

Reply:
😂 Yes. The name or the scope don't provide feedback if we an API user tries to change those after publication. The API just ignores it. I prefer the route of providing feedback that they're trying a bad request, but I think this was introduced way back then, so maybe it is for intended those.
Anyways, thanks for the great and insightful discussion as always, @carolmariaabb! 💯